libcontainer/dmz/README.md

```diff
@@ -15,4 +15,20 @@ It also support all the architectures we support in runc.
 
 If the GOARCH we use for compiling doesn't support nolibc, it fallbacks to using the C stdlib.
 
+## SELinux compatibility issue and a workaround
+
+Older SELinux policy can prevent runc to execute the dmz binary. The issue is
+fixed in [container-selinux v2.224.0].
+
+Distributions that enable SELinux enforced mode and do not have the above fix
+to container-selinux need to build runc with the `runc_dmz_selinux_compat`
+build flag.
```

Comment:
I still prefer a single golden binary that works on any distro.

Reply:
Are you talking about the binaries that we release? If yes, we can definitely build them with the `runc_dmz_selinux_compat` enabled. Please see the last commit (just added).

Reply:
Not just binaries we release. Third parties often distribute runc binaries they built from the source by themselves, and they may overlook this change when they bump up runc. 

So I suggest enabling the SELinux workaround by default without an extra tag. This sacrifies memory footprint on  SELinux-based distros, but compatibility matters more.

Reply:
> Not just binaries we release. Third parties often distribute runc binaries they built from the source by themselves, and they may overlook this change when they bump up runc.

And if this happens (hopefully in an -rc release), it might serve as a reminder that they need to update container-selinux. Also, the only distro which is affected is CentOS/RHEL 7, which will go EOL in less than a year.
 
> So I suggest enabling the SELinux workaround by default without an extra tag. This sacrifies memory footprint on SELinux-enabled distros, but compatibility matters more.

OK, what if we reverse the meaning of the flag (and rename it to, say, `runc_dmz_selinux_no_compat`)? So that instead of asking CentOS 7 builds to set `runc_dmz_selinux_compat`, we'll ask CentOS Stream 8-9 and Fedora distros to set `runc_dmz_selinux_no_compat`.

Reply:
@AkihiroSuda implemented as per ^^^; PTAL